crates/core/tests/operations.rs

```diff
@@ -2822,3 +2822,140 @@ async fn test_update_no_change_notification() -> TestResult {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_proximity_cache_query() -> TestResult {
```

Comment:
this seems a bit insufficient for testing this out, wouldnt we want multiple peers connecting and forming a small network so we see the notifications spread through the network?

we can expose the query information via ws to observe the internal state of each node to assert that indeed we get expectd notifications